tm2/pkg/crypto/keys/client/broadcast.go

```diff
@@ -121,6 +123,9 @@ func BroadcastHandler(cfg *BroadcastCfg) (*ctypes.ResultBroadcastTxCommit, error
 		res, err := SimulateTx(cli, bz)
 		hasError := err != nil || res.CheckTx.IsErr() || res.DeliverTx.IsErr()
 		if cfg.DryRun || hasError {
+			if cfg.DryRun { // we estmate the gas fee in dry run
+				err = estimateGasFee(cli, res)
+			}
 			return res, err
 		}
```

Comment:
```suggestion
		hasError := err != nil || res.CheckTx.IsErr() || res.DeliverTx.IsErr()
		if hasError {
			return res, err
		}
		if cfg.DryRun {
			err = estimateGasFee(cli, res)
			return res, err
		}
```

If we don't do that we are swallowing an error from before.

Reply:
good catch!